packages/aft/lib/src/commands/test/unit_test_command.dart

```diff
@@ -0,0 +1,187 @@
+        }
+      }
+
+      await _executeTest(package, folio);
+    }
+    printResults(
+      folio,
+      verbose: verbose,
+    );
+  }
+
+  Future<void> _executeTest(
+    PackageInfo package,
+    TestReportFolio folio,
+  ) async {
+    final completer = Completer<void>();
+    switch (package.flavor) {
+      case PackageFlavor.flutter:
+        flutterTest(
+          workingDirectory: package.path,
+          // runInShell required for Windows
+          runInShell: true,
+        ).listen(
+          _onData(
+            package,
+            folio,
+            completer,
+          ),
+        );
+        break;
+      case PackageFlavor.dart:
+        dartTest(
+          workingDirectory: package.path,
+        ).listen(
+          _onData(
+            package,
+            folio,
+            completer,
+          ),
+        );
+        break;
+    }
+    await completer.future;
+  }
+
+  void Function(TestEvent) _onData(
+    PackageInfo package,
+    TestReportFolio folio,
+    Completer<void> completer,
+  ) {
+    return (TestEvent event) {
+      switch (event.runtimeType) {
+        case TestStartEvent:
+          final testStartEvent = event as TestStartEvent;
+          final url = testStartEvent.test.url ?? testStartEvent.test.rootUrl;
+          if (url is String) {
+            stdout.writeln(
+              'testing ${package.name}: ${testStartEvent.test.name}...',
+            );
+            final fileName = p.basename(url);
+            folio.reportByFile(package, fileName)?.testId =
+                testStartEvent.test.id;
+          }
+          break;
+        case ErrorTestEvent:
+          final errorTestEvent = event as ErrorTestEvent;
+          if (errorTestEvent.isFailure) {
+            folio.reportByTestId(package, errorTestEvent.testID)?.failures.add(
+                  '${formatException('Package')}: ${package.name}\nStackTrace: ${errorTestEvent.stackTrace}\n',
+                );
+          } else {
+            folio
+                .reportByTestId(package, errorTestEvent.testID)
+                ?.exceptions
+                .add(
+                  '* An Exception occurred in ${package.name} test and was not a test failure.\n  ${errorTestEvent.error}\n',
+                );
+          }
+          break;
+
+        case TestDoneEvent:
+          final testDoneEvent = event as TestDoneEvent;
+          final report = folio.reportByTestId(package, testDoneEvent.testID);
+          if (report != null) {
+            _scoreTest(
+              testDoneEvent,
+              report as TestReportScored,
+            );
+          }
+          break;
+        case ExitTestEvent:
+          final exitTestEvent = event as ExitTestEvent;
+          if (exitTestEvent.exitCode != 0) {
+            folio.packagesWithExitExceptions.add(package);
+          }
+          completer.complete();
+          break;
+      }
+    };
+  }
+
+  void _scoreTest(TestDoneEvent event, TestReportScored testReport) {
```

Comment:
I am seeing 0 passed, 0 failed, 0 skipped when I run the Authenticator tests. I do see the golden tests execute though. It seems like the results are not being aggregated correctly.